### PR TITLE
support EFM32PG22C200F512IM40 SoC

### PR DESCRIPTION
**Pull request by HaimYosefHaim**
Hi, to avoid updating the SDK version, I used the same version as in  #19. I just added the relevant files of this SoC.

### CODE REVIEW

File: gecko/Device/SiliconLabs/EFM32PG22/Source/startup_efm32pg22.c [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+*           Device EFM32PG22
+* @version  V2.1.0
+* @date     16. December 2020
+*******************************************************************************
+* # License
+*
+* The licensor of this software is Silicon Laboratories Inc. Your use of this
+* software is governed by the terms of Silicon Labs Master Software License
+* Agreement (MSLA) available at
+* www.silabs.com/about-us/legal/master-software-license-agreement. This
```
Comment (fkokosinski) [inline gecko/Device/SiliconLabs/EFM32PG22/Source/startup_efm32pg22.c:13]: Note that only OSI-compliant licenses are allowed in Zephyr modules.

**Review by fkokosinski — CHANGES_REQUESTED** [review]
The licensing status of the files in the `gecko/Device/SiliconLabs/EFM32PG22/Source/` is unclear to me, as there are two licenses mentioned in them (MSLA and Zlib). Are these files necessary for the SoC to be properly supported in Zephyr?

File: gecko/Device/SiliconLabs/EFM32PG22/Source/startup_efm32pg22.c (hunk 1 above)
Comment (HaimYosefHaim) [reply gecko/Device/SiliconLabs/EFM32PG22/Source/startup_efm32pg22.c]: Sorry. I removed unnecessary for current support in EFM32PG_DK2503A board I want to add

**Comment by fkokosinski** [comment]
Hi @HaimYosefHaim, thanks for cleaning up the licensing status of this PR.

Usually when changes are made to a Zephyr HAL module repository, there exists an associated PR to the main Zephyr repo, so that maintainers can see how the changes in the HAL repository affect/are used in Zephyr itself. For an example, you could refer to https://github.com/zephyrproject-rtos/zephyr/pull/50352.

It doesn't seem you have created one. Since you are adding support for EFM32PG22 SoCs, I assume you want to add a new board to Zephyr. Could you create a PR in Zephyr, so that I can verify how you are using files provided in this PR?

**Comment by jerome-pouiller** [comment]
Support for EFM32PG_DK2503A has never been submitted to the Zephyr repo. We can't merge this PR without the associated PR in Zephyr.

**Pull request closed without merging**